docs/reference/analysis/analyzers/lang-analyzer.asciidoc

```diff
@@ -36,1481 +39,515 @@ following types are supported:
+being stemmed. 
 
-The built-in language analyzers can be reimplemented as `custom` analyzers
-(as described below) in order to customize their behaviour.
+[source,js]
+--------------------------------------------------------------------------------
+include::../../snippets/analysis/configure-english-analyzer-keywords.json[]
+--------------------------------------------------------------------------------
+// SENSE: analysis/configure-english-analyzer-keywords.json
 
-NOTE: If you do not intend to exclude words from being stemmed (the
-equivalent of the `stem_exclusion` parameter above), then you should remove
-the `keyword_marker` token filter from the custom analyzer configuration.
+Using this analyzer to process the string  "There is no way that writers can be 
+tamed and rendered civilized" generates the following tokens:
+
+`there, is, no, way, that, writer, can, be, tame, and, render, civilized`
+
+[source,js]
+--------------------------------------------------------------------------------
+include::../../snippets/analysis/my-english-analyzer.json[]
+--------------------------------------------------------------------------------
+// SENSE: analysis/my-english-analyzer.json
+
+==== Customizing Language Analyzers
+
+You can reimplement the built-in language analyzers as <<analysis-custom-analyzer, `custom`>> analyzers
+if you need to alter their behaviour. The following examples show how
+to replicate each language analyzer.
+
+NOTE: If you do not intend to exclude words from being stemmed, 
+omit the `keyword_marker` token filter from your custom analyzer configuration.
 
 [[arabic-analyzer]]
-===== `arabic` analyzer
+===== Arabic Analyzer
 
-The `arabic` analyzer could be reimplemented as a `custom` analyzer as follows:
+The following example shows how to reimplement the  `arabic` analyzer
+as a `custom` analyzer. You can omit the `keyword_marker` filter if you 
+don't need to exclude words from stemming.
 
 [source,js]
-----------------------------------------------------
-{
-  "settings": {
-    "analysis": {
-      "filter": {
-        "arabic_stop": {
-          "type":       "stop",
-          "stopwords":  "_arabic_" <1>
-        },
-        "arabic_keywords": {
-          "type":       "keyword_marker",
-          "keywords":   [] <2>
-        },
-        "arabic_stemmer": {
-          "type":       "stemmer",
-          "language":   "arabic"
-        }
-      },
-      "analyzer": {
-        "arabic": {
-          "tokenizer":  "standard",
-          "filter": [
-            "lowercase",
-            "arabic_stop",
-            "arabic_normalization",
-            "arabic_keywords",
-            "arabic_stemmer"
-          ]
-        }
-      }
-    }
-  }
-}
-----------------------------------------------------
-<1> The default stopwords can be overridden with the `stopwords`
-    or `stopwords_path` parameters.
-<2> This filter should be removed unless there are words which should
```

Comment:
All of the examples without keywords are broken.  If you actually follow these steps in sense, you end up creating a broken index, because you close, update the settings (with invalid settings, ie no keywords), then open, and the index cannot be reopened (an event which only gets reported in the logs).  

This is why I prefer showing analyzers at index creation time.

The keywords token filter should be removed if there are no keywords listed (as mentioned in the original callouts)